multisense_ros/src/camera.cpp

```diff
@@ -142,12 +145,12 @@ void writePoint(sensor_msgs::PointCloud2 &pointcloud,
     {
         case 8:
         {
-            const uint32_t luma = static_cast<uint32_t>(reinterpret_cast<const uint8_t*>(image.imageDataP)[image_index]);
+            const uint8_t luma = reinterpret_cast<const uint8_t*>(image.imageDataP)[image_index];
             return writePoint(pointcloud, pointcloud_index, point, luma);
         }
         case 16:
         {
-            const uint32_t luma = static_cast<uint32_t>(reinterpret_cast<const uint16_t*>(image.imageDataP)[image_index]);
+            const uint16_t luma = reinterpret_cast<const uint16_t*>(image.imageDataP)[image_index];
             return writePoint(pointcloud, pointcloud_index, point, luma);
         }
```

Comment:
I don't think we actually support this with the pointclouds we initialized. Thinking about the ST21 16bit thermal images here